docs/src/app/components/pages/components/Stepper/VerticalLinearStepperWithOptionalStep.jsx

```diff
@@ -0,0 +1,203 @@
+      return;
+    }
+
+    this.setState({
+      activeStepIndex: stepIndex,
+      lastActiveStepIndex: Math.max(lastActiveStepIndex, activeStepIndex),
+    });
+  },
+
+  updateCompletedSteps(stepIndex) {
+    return this.state.statusSteps[stepIndex];
+  },
+
+  continue() {
+    const {
+      activeStepIndex,
+      lastActiveStepIndex,
+      statusSteps,
+    } = this.state;
+
+    statusSteps[activeStepIndex] = true;
+
+    this.setState({
+      activeStepIndex: activeStepIndex + 1,
+      statusSteps: statusSteps,
+      lastActiveStepIndex: Math.max(lastActiveStepIndex, activeStepIndex + 1),
+    });
+  },
+
+  render() {
+    return (
+      <Paper style={{width: 500, margin: 'auto'}}>
+        <div style={{
+          textAlign: 'center',
+          padding: 10,
+          fontSize: 20,
+        }}
+        >
+          Linear Stepper with optional steps
+        </div>
+        <Stepper
+          activeStepIndex={this.state.activeStepIndex}
+          onHeaderStepTouch={this.selectStep}
+          completedIcon={
+            <FontIcon className="material-icons" style={{fontSize: 14}}>
+              done
+            </FontIcon>
+          }
+          updateCompletedStatusOfStep={this.updateCompletedSteps}
+        >
+          <Step
+            orderStepLabel="1"
+            stepLabel="Stand in front of door"
+            controlButtonsGroup={[
+              <RaisedButton key={0} label="Continue" primary={true}
+                onClick={this.continue}
+              />,
+              <FlatButton key={1} label="Cancel" />,
+            ]}
+          >
+            <div style={{height: 200}}>
+              My name is Nam
+            </div>
+          </Step>
+          <Step
+            orderStepLabel="2"
+            stepLabel="Open the door"
+            controlButtonsGroup={[
+              <RaisedButton key={0} label="Continue" primary={true}
+                onClick={this.continue}
+              />,
+              <FlatButton key={1} label="Cancel" />,
+            ]}
+          >
+            <div style={{height: 50}}>
+              Get out of my eyes
+            </div>
+          </Step>
+
+          <Step
+            orderStepLabel="3"
+            stepLabel="Find the key to open the door"
+            controlButtonsGroup={[
+              <RaisedButton key={0} label="Finish" primary={true}
+                onClick={this.continue}
+              />,
+              <FlatButton key={1} label="Cancel" />,
+            ]}
+          >
+            <div style={{height: 50}}>
+              Good bye
+            </div>
+          </Step>
+
+          <Step
+            orderStepLabel="4"
+            stepLabel="Find and find"
+            isCompleted={false}
+            optional={true}
+            stepLabelView={
+              <div>
+                <div>Dock the door</div>
+                <div style={{fontSize: 10, lineHeight: '5px'}}>optional</div>
```

Comment:
We might have to find a way to make this composable/configurable for i18n.

@oliviertassinari @alitaheri @mbrookes Thoughts on how we should handle this `'optional'` word?